ui/components/Modals/Information/infoModal.js

```diff
@@ -0,0 +1,365 @@
+import { useSnackbar } from 'notistack';
+import Filter from '../../../public/static/img/drawer-icons/filter_svg.js';
+import { PATTERN_ENDPOINT, FILTER_ENDPOINT } from '../../../constants/endpoints';
+import { useNotification } from '../../../utils/hooks/useNotification';
+import { EVENT_TYPES } from '../../../lib/event-types';
+import axios from 'axios';
+import _ from 'lodash';
+import RJSFWrapper from '../../MesheryMeshInterface/PatternService/RJSF_wrapper';
+import dataFetch from '../../../lib/data-fetch';
+import CircularProgress from '@mui/material/CircularProgress';
+
+const APPLICATION_PLURAL = 'applications';
+const FILTER_PLURAL = 'filters';
+const PATTERN_PLURAL = 'patterns';
+
+const InfoModal = (props) => {
+  const {
+    infoModalOpen,
+    handleInfoModalClose,
+    resourceOwnerID, //id to get data of user to get name of that person
+    dataName,
+    selectedResource,
+    currentUserID, // id of current user
+    formSchema,
+  } = props;
+
+  // const theme = uiTheme.theme;
+  const formRef = React.createRef();
+  const [formState, setFormState] = useState(selectedResource?.catalog_data || {});
+  const [isCatalogDataEqual, setIsCatalogDataEqual] = useState(false);
+  const [resourceUserProfile, setResourceUserProfile] = useState(null);
+  const [saveFormLoading, setSaveFormLoading] = useState(false);
+  const [uiSchema, setUiSchema] = useState({});
+  const { notify } = useNotification();
+  const theme = useTheme();
+
+  const classes = useStyles();
+  const formatDate = (date) => {
+    return moment(date).utc().format('MMMM Do YYYY, h:mm:ss A');
+  };
+  const { enqueueSnackbar } = useSnackbar();
+
+  const handleCopy = () => {
+    // navigator.clipboard.writeText(getSharableCommonHostAndprotocolLink(selectedResource));
+    enqueueSnackbar(`Link to "${selectedResource.name}" is copied to clipboard`, {
+      variant: 'info',
+      autoHideDuration: 2000,
+    });
+  };
+
+  const handleSubmit = () => {
+    if (formRef.current && formRef.current.validateForm()) {
+      setSaveFormLoading(true);
+      let body = null;
+      if (dataName === PATTERN_PLURAL) {
+        body = {
+          body: JSON.stringify({
+            pattern_data: {
+              catalog_data: formState,
+              id: selectedResource.id,
+            },
+          }),
+        };
+      } else if (dataName === FILTER_PLURAL) {
+        setSaveFormLoading(true);
+        body = {
+          body: JSON.stringify({
+            filter_data: {
+              catalog_data: formState,
+              id: selectedResource.id,
+            },
+          }),
+        };
+      }
+
+      axios
+        .post(dataName === PATTERN_PLURAL ? PATTERN_ENDPOINT : FILTER_ENDPOINT, body)
+        .then(() => {
+          setSaveFormLoading(false);
+          notify({
+            message: `${selectedResource.name} data saved successfully`,
+            event_type: EVENT_TYPES.SUCCESS,
+          });
+        })
```

Comment:
@theBeginner86 this is how I'm sending catalog data

Reply:
Looking into it now....